karaf/osgi-jta/src/main/java/org/jboss/narayana/quickstart/osgi/CommitExample.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2013, Red Hat, Inc. and/or its affiliates, and individual
+ * contributors by the @authors tag. See the copyright.txt in the
+ * distribution for a full listing of individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.narayana.quickstart.osgi;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.jboss.narayana.quickstart.xa.DummyXAResource;
+
+import javax.transaction.Transaction;
+import javax.transaction.TransactionManager;
+
+/**
+ * @author <a href="mailto:zfeng@redhat.com">Amos Feng</a>
+ */
+
+@Command(scope = "narayana-quickstart", name = "testCommit", description = "The JTA Commit Example")
+@Service
+public class CommitExample implements Action {
+    @Reference
+    private TransactionManager tm;
+
+    @Override
+    public Object execute() throws Exception {
+        System.out.println("testCommit");
+
+        if (tm == null) {
+            System.out.println("Can not get transaction manager");
+        } else {
+            DummyXAResource xa1 = new DummyXAResource();
+            DummyXAResource xa2 = new DummyXAResource();
+
+            tm.begin();
+            Transaction transaction = tm.getTransaction();
+            transaction.enlistResource(xa1);
```

Comment:
Is this the recommended way in Karaf - do they not have an equivalent of JCA I wonder?

I guess for the dummyxaresource it has to be this way but can we try a H2 datasource somehow?